### PR TITLE
JNG-5838 transform primitive fields

### PR DESCRIPTION
**Pull request by noherczeg**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5838" title="JNG-5838" target="_blank">Sub-taskJNG-5838</a>  Transform primitive fields
  </td></table>
  <br />

### CODE REVIEW

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        return self.equivalent("CreateDerivedTransferAttribute");
+    }
+    if (self.maps()) {
+        return self.equivalent("CreateMappedTransferAttribute");
+    }
+    throw "Could not determinate attribute type: " + s.fqName();
+}
+
+@cached
+operation JSL!TransferFieldDeclaration isSortable() : Boolean {
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol:54]: I believe enums are sortable as well, hence the name "ordinal"

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        self.referenceType.`primitive` == "numeric"  or
+        self.referenceType.`primitive` == "date" or
+        self.referenceType.`primitive` == "timestamp" or
+        self.referenceType.`primitive` == "time" or
+        self.referenceType.`primitive` == "boolean"
+    );
+}
+
+@cached
+operation JSL!TransferFieldDeclaration isFilterable() : Boolean {
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol:69]: Enum filter?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol (hunk 1 above)
Comment (bencegelei) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol:54]: sry, i f****d up the selected code part :D

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol (hunk 2 above)
Comment (bencegelei) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol:69]: sry, i f****d up the selected code part :D

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol (hunk 1 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol:54]: This was copied from jsl2psm, so we'll keep it like this for now. Once we start E2E testing the JSL stack, we can correct it if necessary.

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol (hunk 2 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/transferFieldDeclaration.eol:69]: This was copied from jsl2psm, so we'll keep it like this for now. Once we start E2E testing the JSL stack, we can correct it if necessary.

**Review by bencegelei — APPROVED** [review]

**Review by Madmat8 — APPROVED** [review]

**Pull request merged**